tests/v2/test_waku_store_queue.nim

```diff
@@ -0,0 +1,269 @@
+{.used.}
+
+import
+  std/sequtils,
+  testutils/unittests,
+  ../../waku/v2/protocol/waku_store/waku_store_types
+
+procSuite "Sorted store queue":
+
+  # Helper functions
+  proc genIndexedWakuMessage(i: int8): IndexedWakuMessage =
+    ## Use i to generate an IndexedWakuMessage
+    var data {.noinit.}: array[32, byte]
+    for x in data.mitems: x = i.byte
+    return IndexedWakuMessage(msg: WakuMessage(payload: @[byte i], timestamp: float64(i)),
+                              index: Index(receiverTime: float64(i), senderTime: float64(i), digest: MDigest[256](data: data)))
+
+  # Test variables  
+  let
+    capacity = 5
+    unsortedSet = [5,1,3,2,4]
+  
+  var testStoreQueue = StoreQueueRef.new(capacity)
+  for i in unsortedSet:
+    testStoreQueue.add(genIndexedWakuMessage(i.int8))
+  
+  test "Store queue can be created with limited capacity":
+    var stQ = StoreQueueRef.new(capacity)
+    check:
+      stQ.len == 0 # Empty when initialised
+    
+    for i in 1..capacity: # Fill up the queue
+      stQ.add(genIndexedWakuMessage(i.int8))
+    
+    check:
+      stQ.len == capacity
+    
+    # Add one more. Capacity should not be exceeded.
+    stQ.add(genIndexedWakuMessage(capacity.int8 + 1))
+    
+    check:
+      stQ.len == capacity
+  
+  test "Store queue sort-on-insert works":    
+    # Walk forward through the set and verify ascending order
+    var prevSmaller = genIndexedWakuMessage(min(unsortedSet).int8 - 1).index
+    for i in testStoreQueue.fwdIterator:
+      let (index, indexedWakuMessage) = i
+      check cmp(index, prevSmaller) > 0
+    
+    # Walk backward through the set and verify descending order
+    var prevLarger = genIndexedWakuMessage(max(unsortedSet).int8 + 1).index
+    for i in testStoreQueue.bwdIterator:
+      let (index, indexedWakuMessage) = i
+      check cmp(index, prevLarger) < 0
```

Comment:
Shouldn't `prevLarger` gets updated like: (the same for the loop above)
```suggestion
      check cmp(index, prevLarger) < 0
      prevLarger = index
```

Reply:
Yup! Good catch. :) [a16c814](https://github.com/status-im/nim-waku/pull/849/commits/a16c814eb57921443115d41e199bb35eaf7ef546)